### PR TITLE
[MDCPageControl][MDCBottomAppBarView] Strong/Weak Self references inside block.

### PR DESCRIPTION
**Pull request by mohammadcazig**
change strong references to weak references so we don't hold on to the components more than we need too.
#2934

### CODE REVIEW

File: components/BottomAppBar/src/MDCBottomAppBarView.m [hunk 1]
```diff
@@ -323,10 +323,11 @@ - (void)setFloatingButtonElevation:(MDCBottomAppBarFloatingButtonElevation)float
   }
   if (animated) {
     [_floatingButton setElevation:1 forState:UIControlStateNormal];
+    __weak typeof(self) weakSelf = self;
     dispatch_after(dispatch_time(DISPATCH_TIME_NOW, kMDCButtonAnimationDuration * NSEC_PER_MSEC),
                    dispatch_get_main_queue(), ^{
-                     [self insertSubview:self.floatingButton atIndex:subViewIndex];
-                     [self.floatingButton setElevation:elevation forState:UIControlStateNormal];
+                     [weakSelf insertSubview:weakSelf.floatingButton atIndex:subViewIndex];
```
Comment (romoore) [inline components/BottomAppBar/src/MDCBottomAppBarView.m:329]: I think you would want to strongify here.  Calling `-insertSubview:atIndex:` cannot accept a `nil` view.  Even though I believe the right hand side is evaluated first (leaving the receiver `nil`), it still feels like a risk.

File: components/PageControl/src/MDCPageControl.m [hunk 2]
```diff
@@ -171,16 +171,18 @@ - (void)setCurrentPage:(NSInteger)currentPage
       // We are using the delay to increase the time between the end of the extension of the track
       // ahead of the dots movement and the contraction of the track under the dot at the
       // destination.
+      __weak typeof(self) weakSelf = self;
```
Comment (romoore) [inline components/PageControl/src/MDCPageControl.m:174]: Can you write a unit test to cover this behavior before/after?

File: components/PageControl/src/MDCPageControl.m [hunk 3]
```diff
@@ -264,16 +266,21 @@ - (void)scrollViewDidScroll:(UIScrollView *)scrollView {
     // If the animation block has a delay it translates to the beginTime of the CAAnimation. We need
     // to ensure that we delay our animation of the page control to keep in sync with the animation
     // of the scrollView.contentOffset.
+    __weak typeof(self) weakSelf = self;
```
Comment (romoore) [inline components/PageControl/src/MDCPageControl.m:269]: Please add a unit test for this code as well.

**Review by romoore — CHANGES_REQUESTED** [review]

**Comment by mohammadcazig** [comment]
I feel like this change does not give us a lot of benefits and it might includes some possible risks. So we decided not to make the change.

**Pull request closed without merging**